<?xml version="1.0"?>
<!--
Software License Agreement (BSD)

\file      server_service_shims.launch
\authors   Dave Niewinski <dniewinski@clearpathrobotics.com>
\copyright Copyright (c) 2017, Clearpath Robotics, Inc., All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that
the following conditions are met:
 * Redistributions of source code must retain the above copyright notice, this list of conditions and the
   following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
   following disclaimer in the documentation and/or other materials provided with the distribution.
 * Neither the name of Clearpath Robotics nor the names of its contributors may be used to endorse or promote
   products derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WAR-
RANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, IN-
DIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
  <node name="configure_ark_servicer" pkg="ark_bridge" type="arkconfigsettings_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/ark_config_settings_call"/>
    <param name="response_topic" value="/ark_bridge/ark_config_settings_response"/>
    <param name="service_name" value="/configure_ark"/>
  </node>

  <node name="map_data_clear_servicer" pkg="ark_bridge" type="empty_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/map_data_clear_call"/>
    <param name="response_topic" value="/ark_bridge/map_data_clear_response"/>
    <param name="service_name" value="/map_data/clear"/>
  </node>
  <node name="control_selection_set_mode_servicer" pkg="ark_bridge" type="setautonomymode_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_set_mode_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_set_mode_response"/>
    <param name="service_name" value="/control_selection/set_mode"/>
  </node>
  <node name="slam_set_mode_servicer" pkg="ark_bridge" type="setslammode_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/slam_set_mode_call"/>
    <param name="response_topic" value="/ark_bridge/slam_set_mode_response"/>
    <param name="service_name" value="/slam/set_mode"/>
  </node>
  <node name="slam_set_initial_pose_servicer" pkg="ark_bridge" type="setpose_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/slam_set_initial_pose_call"/>
    <param name="response_topic" value="/ark_bridge/slam_set_initial_pose_response"/>
    <param name="service_name" value="/slam/set_initial_pose"/>
  </node>
  <node name="control_selection_autonomy_clear_all_disabled_servicer" pkg="ark_bridge" type="empty_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_autonomy_clear_all_disabled_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_autonomy_clear_all_disabled_response"/>
    <param name="service_name" value="/control_selection/autonomy/clear_all_disabled"/>
  </node>
  <node name="control_selection_autonomy_clear_all_paused_servicer" pkg="ark_bridge" type="empty_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_autonomy_clear_all_paused_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_autonomy_clear_all_paused_response"/>
    <param name="service_name" value="/control_selection/autonomy/clear_all_paused"/>
  </node>
  <node name="control_selection_autonomy_disable_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_autonomy_disable_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_autonomy_disable_response"/>
    <param name="service_name" value="/control_selection/autonomy/disable"/>
  </node>
  <node name="control_selection_autonomy_enable_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_autonomy_enable_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_autonomy_enable_response"/>
    <param name="service_name" value="/control_selection/autonomy/enable"/>
  </node>
  <node name="control_selection_autonomy_pause_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_autonomy_pause_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_autonomy_pause_response"/>
    <param name="service_name" value="/control_selection/autonomy/pause"/>
  </node>
  <node name="control_selection_autonomy_resume_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_autonomy_resume_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_autonomy_resume_response"/>
    <param name="service_name" value="/control_selection/autonomy/resume"/>
  </node>
  <node name="control_selection_core_clear_all_disabled_servicer" pkg="ark_bridge" type="empty_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_core_clear_all_disabled_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_core_clear_all_disabled_response"/>
    <param name="service_name" value="/control_selection/core/clear_all_disabled"/>
  </node>
  <node name="control_selection_core_clear_all_paused_servicer" pkg="ark_bridge" type="empty_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_core_clear_all_paused_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_core_clear_all_paused_response"/>
    <param name="service_name" value="/control_selection/core/clear_all_paused"/>
  </node>
  <node name="control_selection_core_disable_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_core_disable_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_core_disable_response"/>
    <param name="service_name" value="/control_selection/core/disable"/>
  </node>
  <node name="control_selection_core_enable_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_core_enable_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_core_enable_response"/>
    <param name="service_name" value="/control_selection/core/enable"/>
  </node>
  <node name="control_selection_core_pause_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_core_pause_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_core_pause_response"/>
    <param name="service_name" value="/control_selection/core/pause"/>
  </node>
  <node name="control_selection_core_resume_servicer" pkg="ark_bridge" type="string_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/control_selection_core_resume_call"/>
    <param name="response_topic" value="/ark_bridge/control_selection_core_resume_response"/>
    <param name="service_name" value="/control_selection/core/resume"/>
  </node>
  <node name="map_data_delete_map_servicer" pkg="ark_bridge" type="deletemap_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/map_data_delete_map_call"/>
    <param name="response_topic" value="/ark_bridge/map_data_delete_map_response"/>
    <param name="service_name" value="/map_data/delete_map"/>
  </node>
  <node name="map_data_list_maps_servicer" pkg="ark_bridge" type="listmaps_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/map_data_list_maps_call"/>
    <param name="response_topic" value="/ark_bridge/map_data_list_maps_response"/>
    <param name="service_name" value="/map_data/list_maps"/>
  </node>
  <node name="map_data_load_map_from_disk_servicer" pkg="ark_bridge" type="loadmapfromdisk_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/map_data_load_map_from_disk_call"/>
    <param name="response_topic" value="/ark_bridge/map_data_load_map_from_disk_response"/>
    <param name="service_name" value="/map_data/load_map_from_disk"/>
  </node>
  <node name="map_data_save_map_to_disk_servicer" pkg="ark_bridge" type="savemaptodisk_service" output="screen" respawn="true">
    <param name="call_topic" value="/ark_bridge/map_data_save_map_to_disk_call"/>
    <param name="response_topic" value="/ark_bridge/map_data_save_map_to_disk_response"/>
    <param name="service_name" value="/map_data/save_map_to_disk"/>
  </node>
</launch>
